src/averageAges.js

```diff
@@ -54,8 +71,30 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let children;
+
+  if (onlyWithSon) {
+    children = people.filter((person) => person.sex === 'm');
+  } else {
+    children = people.filter((person) => person.mother !== undefined);
+  }
+
+  const diffs = children.map((child) => {
+    const mother = people.find((person) => person.name === child.mother);
+
+    if (mother !== undefined) {
+      return child.born - mother.born;
+    }
+  }).filter((diff) => diff !== undefined);
```

Comment:
Instead of using separate map and filter functions, you can use a single reduce function to calculate the sum of age differences and the number of differences at once, then calculate the average outside the reduce function.